rust/lance-core/src/io/writer/statistics.rs

```diff
@@ -346,12 +968,91 @@ impl StatisticsBuilder {
 #[cfg(test)]
 mod tests {
     use arrow_array::{
-        BooleanArray, Date32Array, Float32Array, Int32Array, Int64Array, StructArray,
+        builder::StringDictionaryBuilder, BinaryArray, BooleanArray, Date32Array, Date64Array,
+        Decimal128Array, DictionaryArray, DurationMicrosecondArray, DurationMillisecondArray,
+        DurationNanosecondArray, DurationSecondArray, FixedSizeBinaryArray, Float32Array,
+        Float64Array, Int16Array, Int32Array, Int64Array, Int8Array, LargeBinaryArray,
+        LargeListArray, LargeStringArray, ListArray, StringArray, StructArray,
+        Time32MillisecondArray, Time32SecondArray, Time64MicrosecondArray, Time64NanosecondArray,
+        TimestampMicrosecondArray, TimestampMillisecondArray, TimestampNanosecondArray,
+        TimestampSecondArray, UInt16Array, UInt32Array, UInt64Array, UInt8Array,
     };
 
-    use arrow_schema::{Field as ArrowField, Schema as ArrowSchema};
-
     use super::*;
+    use arrow_schema::{Field as ArrowField, Fields as ArrowFields, Schema as ArrowSchema};
+
+    #[test]
+    fn test_edge_cases() {
+        // No arrays, datatype can't be inferred
+        let arrays: Vec<ArrayRef> = vec![];
+        let array_refs = arrays.iter().collect::<Vec<_>>();
```

Comment:
I think we'd rather panic here actually. For types where we don't have null support (integers), we don't want to accidentally have nulls for min / max.

Reply:
Makes sense! Changed.